packages/auto-approve/test/check-config.test.ts

```diff
@@ -0,0 +1,300 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  validateYaml,
+  validateSchema,
+  checkCodeOwners,
+} from '../src/check-config.js';
+import {describe, it} from 'mocha';
+import assert from 'assert';
+import * as fs from 'fs';
+import yaml from 'js-yaml';
+import nock from 'nock';
+const {Octokit} = require('@octokit/rest');
+
+const octokit = new Octokit({
+  auth: 'mypersonalaccesstoken123',
+});
+const CONFIGURATION_FILE_PATH = 'auto-approve.yml';
+
+nock.disableNetConnect();
+
+function getCodeOwnersFile(response: string | undefined, status: number) {
+  return nock('https://api.github.com')
+    .get('/repos/owner/repo/contents/.github%2FCODEOWNERS')
+    .reply(
+      status,
+      response ? {content: Buffer.from(response).toString('base64')} : undefined
+    );
+}
+
+async function invalidateSchema(configNum: number) {
+  return await validateSchema(
+    yaml.load(
+      fs.readFileSync(
+        `./test/fixtures/config/invalid-schemas/invalid-schema${configNum}.yml`,
+        'utf8'
+      )
+    )
+  );
+}
+
+describe('check for config', () => {
```

Comment:
Kudos for being thorough.  With all these tests, I'm very confident this code will behave as expected.